src/SWP/Bundle/ContentBundle/Service/RouteToArticleMapper.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/**
+ * This file is part of the Superdesk Web Publisher Content Bundle.
+ *
+ * Copyright 2016 Sourcefabric z.ú. and contributors.
+ *
+ * For the full copyright and license information, please see the
+ * AUTHORS and LICENSE files distributed with this source code.
+ *
+ * @copyright 2016 Sourcefabric z.ú.
+ * @license http://www.superdesk.org/license
+ */
+namespace SWP\Bundle\ContentBundle\Service;
+
+use Doctrine\ODM\PHPCR\DocumentRepository;
+use Doctrine\ORM\EntityRepository;
+use Psr\Log\LoggerInterface;
+use SWP\Bundle\ContentBundle\Model\ArticleInterface;
+use SWP\Bundle\ContentBundle\Model\RouteToArticle;
+use Symfony\Component\ExpressionLanguage\ExpressionLanguage;
+
+class RouteToArticleMapper
+{
+    /**
+     * @var EntityRepository
+     */
+    private $routeToArticleRepository;
+
+    /**
+     * @var DocumentRepository
+     */
+    private $routeRepository;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @var ExpressionLanguage
+     */
+    private $language;
+
+    public function __construct(EntityRepository $routeToArticleRepository, DocumentRepository $routeRepository, LoggerInterface $logger)
+    {
+        $this->routeToArticleRepository = $routeToArticleRepository;
+        $this->routeRepository = $routeRepository;
+        $this->logger = $logger;
+        $this->language = new ExpressionLanguage();
```

Comment:
This seems like over-engineering to me. If we really want to decouple this feature from expression language, we will also have to make it possible to override the form used to create and update a RouteToArticleEntity. I really think we should concentrate of keeping things as simple as possible in the short term. We can make the code more flexible later, if and when it needs to be. I think we are more likely to discourage possible contributors by overengineering than to encourage them by making it possible override every feature of the code.